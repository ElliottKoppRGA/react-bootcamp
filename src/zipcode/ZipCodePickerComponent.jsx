import React from 'react';
import propTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

const ZipCodePickerComponent = ({ zipcodeChanged, zipcode }) => (
  <div>
    <TextField placeholder="Enter a zip code" onChange={zipcodeChanged} value={zipcode} />
    <Button variant="contained" color="primary">
      Get Weather
    </Button>
    <p>The zipcode is: {zipcode}</p>
  </div>
);

ZipCodePickerComponent.displayName = 'ZipCodePickerComponent';

ZipCodePickerComponent.propTypes = {
  zipcodeChanged: propTypes.func.isRequired,
  zipcode: propTypes.string.isRequired
}

export default ZipCodePickerComponent;
