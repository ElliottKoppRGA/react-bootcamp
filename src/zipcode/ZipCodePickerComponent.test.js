import React from "react";
import { render, cleanup } from "@testing-library/react";
import ZipCodePickerComponent from "./ZipCodePickerComponent";

describe("ZipCodePickerComponent", () => {
  afterEach(cleanup);
  it("should render zipcode", () => {
    const zipcode = "90210";
    const zipcodeChanged = jest.fn();
    const getWeatherClicked = jest.fn();
    const { container } = render(
      <ZipCodePickerComponent
        zipcode={zipcode}
        zipcodeChanged={zipcodeChanged}
        getWeatherClicked={getWeatherClicked}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
