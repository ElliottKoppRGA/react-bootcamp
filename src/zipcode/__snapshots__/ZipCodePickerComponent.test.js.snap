// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZipCodePickerComponent should render zipcode 1`] = `
<div>
  <div>
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          placeholder="Enter a zip code"
          type="text"
          value="90210"
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Get Weather
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <p>
      The zipcode is: 
      90210
    </p>
  </div>
</div>
`;
